### PR TITLE
feat: add link as new web component

### PR DESCRIPTION
**Pull request by chrisdholt**
## Previous Behavior
N/A

## New Behavior
This PR adds link as a new web component.

## Related Issue(s)

- Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: apps/vr-tests-web-components/src/stories/link/link.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
```
Comment (fabricteam) [inline apps/vr-tests-web-components/src/stories/link/link.stories.tsx:1]: ## [🕵 fluentui-web-components-v3 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31676&commitId=158bcaf40dcc43197bfca042659cc36b05a7ebb9&env=prod&clientType=FLUENTUI)


  <details><summary><b>Accordion</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Accordion.Size - Dark Mode.normal.chromium.png|2| Changed|
|Accordion.Size.normal.chromium.png|2| Changed|
|Accordion.Size - RTL.normal.chromium.png|2| Changed|

</details>


  <details><summary><b>Avatar</b> 5 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar.Color.normal.chromium.png|1| Changed|
|Avatar.Active Appearance - Dark Mode.normal.chromium.png|2| Changed|
|Avatar.Active.normal.chromium.png|5| Changed|
|Avatar.Active Appearance.normal.chromium.png|10| Changed|
|Avatar.Color - RTL.normal.chromium.png|1| Changed|

</details>


  <details><summary><b>Badge</b> 6 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Badge.Default.normal.chromium.png|1| Changed|
|Badge.Appearance - Dark Mode.normal.chromium.png|15| Changed|
|Badge.Appearance.normal.chromium.png|1| Changed|
|Badge.Shape.normal.chromium.png|1| Changed|
|Badge.Size.normal.chromium.png|6| Changed|
|Badge.Color.normal.chromium.png|1| Changed|

</details>


  <details><summary><b>Button</b> 57 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Button.Large Icon Only - Dark Mode.default.chromium.png|4| Changed|
|Button.Large With Icon - Dark Mode.pressed.chromium.png|5| Changed|
|Button.Large Icon Only.hover.chromium.png|4| Changed|
|Button.Large Icon Only.default.chromium.png|5| Changed|
|Button.Large Icon Only.pressed.chromium.png|3| Changed|
|Button.Large Icon Only - Dark Mode.hover.chromium.png|3| Changed|
|Button.Large With Icon - RTL.hover.chromium.png|4| Changed|
|Button.Large Icon Only - Dark Mode.pressed.chromium.png|5| Changed|
|Button.Large With Icon.hover.chromium.png|4| Changed|
|Button.Large With Icon - RTL.pressed.chromium.png|3| Changed|
|Button.Large With Icon.pressed.chromium.png|3| Changed|
|Button.Outline Disabled - Dark Mode.default.chromium.png|159| Changed|
|Button.Outline Disabled Focusable - Dark Mode.default.chromium.png|159| Changed|
|Button.Outline Disabled Focusable - Dark Mode.hover.chromium.png|159| Changed|
|Button.Outline Disabled - Dark Mode.pressed.chromium.png|159| Changed|
|Button.Outline Disabled - Dark Mode.hover.chromium.png|159| Changed|
|Button.Large With Icon.default.chromium.png|5| Changed|
|Button.Primary Disabled - Dark Mode.default.chromium.png|44| Changed|
|Button.Primary Disabled Focusable - Dark Mode.default.chromium.png|44| Changed|
|Button.Primary Disabled Focusable.default.chromium.png|15| Changed|
|Button.Primary Disabled Focusable.hover.chromium.png|15| Changed|
|Button.Primary Disabled Focusable - Dark Mode.pressed.chromium.png|44| Changed|
|Button.Outline Disabled Focusable - Dark Mode.pressed.chromium.png|159| Changed|
|Button.Primary Disabled - Dark Mode.hover.chromium.png|44| Changed|
|Button.Primary Disabled Focusable.pressed.chromium.png|15| Changed|
|Button.Primary Disabled.default.chromium.png|15| Changed|
|Button.Primary Disabled Focusable - Dark Mode.hover.chromium.png|44| Changed|
|Button.Primary Disabled.hover.chromium.png|15| Changed|
|Button.Large With Icon - Dark Mode.hover.chromium.png|3| Changed|
|Button.Large With Icon - RTL.default.chromium.png|5| Changed|
|Button.Primary Disabled - Dark Mode.pressed.chromium.png|44| Changed|
|Button.Primary Disabled.pressed.chromium.png|15| Changed|
|Button.Subtle Disabled - Dark Mode.pressed.chromium.png|187| Changed|
|Button.Subtle Disabled Focusable.hover.chromium.png|25| Changed|
|Button.Subtle Disabled Focusable - Dark Mode.default.chromium.png|187| Changed|
|Button.Subtle Disabled.default.chromium.png|25| Changed|
|Button.Subtle Disabled Focusable.default.chromium.png|25| Changed|
|Button.Subtle Disabled Focusable - Dark Mode.pressed.chromium.png|187| Changed|
|Button.Subtle Disabled.pressed.chromium.png|25| Changed|
|Button.Transparent Disabled Focusable - Dark Mode.default.chromium.png|199| Changed|
|Button.Transparent Disabled - Dark Mode.hover.chromium.png|199| Changed|
|Button.Large With Icon - Dark Mode.default.chromium.png|4| Changed|
|Button.Transparent Disabled.hover.chromium.png|27| Changed|
|Button.Transparent Disabled - Dark Mode.default.chromium.png|199| Changed|
|Button.Transparent Disabled Focusable.pressed.chromium.png|27| Changed|
|Button.Transparent Disabled Focusable - Dark Mode.pressed.chromium.png|199| Changed|
|Button.Subtle Disabled Focusable.pressed.chromium.png|25| Changed|
|Button.Subtle Disabled - Dark Mode.hover.chromium.png|187| Changed|
|Button.Subtle Disabled Focusable - Dark Mode.hover.chromium.png|187| Changed|
|Button.Subtle Disabled.hover.chromium.png|25| Changed|
|Button.Transparent Disabled Focusable - Dark Mode.hover.chromium.png|199| Changed|
|Button.Transparent Disabled - Dark Mode.pressed.chromium.png|199| Changed|
|Button.Subtle Disabled - Dark Mode.default.chromium.png|187| Changed|
|Button.Transparent Disabled Focusable.default.chromium.png|27| Changed|
|Button.Transparent Disabled Focusable.hover.chromium.png|27| Changed|
|Button.Transparent Disabled.pressed.chromium.png|27| Changed|
|Button.Transparent Disabled.default.chromium.png|27| Changed|

</details>


  <details><summary><b>Checkbox</b> 26 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Checkbox.Circular - Dark Mode.normal.chromium.png|792| Changed|
|Checkbox.Checked - Dark Mode.normal.chromium.png|982| Changed|
|Checkbox.Circular Disabled - Dark Mode.normal.chromium.png|760| Changed|
|Checkbox.Circular Checked - Dark Mode.normal.chromium.png|1080| Changed|
|Checkbox.Circular Indeterminate - Dark Mode.indeterminate.chromium.png|1102| Changed|
|Checkbox.Circular Disabled.normal.chromium.png|4711| Changed|
|Checkbox.Circular Indeterminate.indeterminate.chromium.png|4869| Changed|
|Checkbox.Circular Indeterminate - Dark Mode.normal.chromium.png|936| Changed|
|Checkbox.Disabled - Dark Mode.normal.chromium.png|739| Changed|
|Checkbox.Disabled Checked - Dark Mode.normal.chromium.png|777| Changed|
|Checkbox.Circular - RTL.normal.chromium.png|4733| Changed|
|Checkbox.Circular Disabled Checked.normal.chromium.png|4728| Changed|
|Checkbox.Indeterminate - Dark Mode.indeterminate.chromium.png|993| Changed|
|Checkbox.Disabled Checked.normal.chromium.png|4721| Changed|
|Checkbox.Disabled.normal.chromium.png|4705| Changed|
|Checkbox.Indeterminate.normal.chromium.png|4776| Changed|
|Checkbox.Circular Checked.normal.chromium.png|4883| Changed|
|Checkbox.Indeterminate - Dark Mode.normal.chromium.png|877| Changed|
|Checkbox.Default - RTL.normal.chromium.png|4751| Changed|
|Checkbox.Circular Indeterminate.normal.chromium.png|4786| Changed|
|Checkbox.Circular Disabled Checked - Dark Mode.normal.chromium.png|799| Changed|
|Checkbox.Indeterminate.indeterminate.chromium.png|4847| Changed|
|Checkbox.Checked.normal.chromium.png|4855| Changed|
|Checkbox.Circular.normal.chromium.png|4734| Changed|
|Checkbox.Default - Dark Mode.normal.chromium.png|814| Changed|
|Checkbox.Default.normal.chromium.png|4751| Changed|

</details>


  <details><summary><b>Divider</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Divider.With Svg.normal.chromium.png|3| Changed|
|Divider.With Svg - RTL.normal.chromium.png|3| Changed|
|Divider.Vertical With Svg.normal.chromium.png|4| Changed|

</details>


  <details><summary><b>Label</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Label.Size Small.normal.chromium.png|661| Changed|
|Label.Required - RTL.normal.chromium.png|11| Changed|

</details>


  <details><summary><b>Link</b> 10 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Link.Default.default.chromium.png|0| Added|
|Link.Default.hover.chromium.png|0| Added|
|Link.Default.pressed.chromium.png|0| Added|
|Link.Inline.default.chromium.png|0| Added|
|Link.Inline.hover.chromium.png|0| Added|
|Link.Inline.pressed.chromium.png|0| Added|
|Link.Subtle.default.chromium.png|0| Added|
|Link.Subtle.hover.chromium.png|0| Added|
|Link.Subtle.pressed.chromium.png|0| Added|
|Link.With Long Text.default.chromium.png|0| Added|

</details>


  <details><summary><b>MenuList</b> 16 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|MenuList.Checkbox - RTL.2nd selected.chromium.png|1| Changed|
|MenuList.Checkbox With Icons - RTL.normal.chromium.png|4| Changed|
|MenuList.Checkbox With Icons - RTL.2nd selected.chromium.png|5| Changed|
|MenuList.Checkbox With Icons.2nd selected.chromium.png|5| Changed|
|MenuList.Checkbox With Icons.normal.chromium.png|4| Changed|
|MenuList.Checkbox With Icons - RTL.1st selected.chromium.png|4| Changed|
|MenuList.Radio With Icons - RTL.1st selected.chromium.png|4| Changed|
|MenuList.Radio With Icons.2nd selected.chromium.png|4| Changed|
|MenuList.Radio With Icons - RTL.2nd selected.chromium.png|4| Changed|
|MenuList.Radio With Icons.normal.chromium.png|4| Changed|
|MenuList.Checkbox.2nd selected.chromium.png|1| Changed|
|MenuList.With Icons.normal.chromium.png|4| Changed|
|MenuList.Radio With Icons.1st selected.chromium.png|4| Changed|
|MenuList.Checkbox With Icons.1st selected.chromium.png|4| Changed|
|MenuList.Radio With Icons - RTL.normal.chromium.png|4| Changed|
|MenuList.With Icons.hover menuitem.chromium.png|3| Changed|

</details>


  <details><summary><b>Slider</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Slider.Size Small.rightArrow.chromium.png|1| Changed|
|Slider.Size Small.normal.chromium.png|1| Changed|

</details>


  <details><summary><b>Switch</b> 34 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Switch.Checked - Dark Mode.normal.chromium.png|1130| Changed|
|Switch.Checked - RTL.hover.chromium.png|5150| Changed|
|Switch.Checked.hover.chromium.png|5150| Changed|
|Switch.Checked - RTL.normal.chromium.png|5132| Changed|
|Switch.Checked.normal.chromium.png|5132| Changed|
|Switch.Default.normal.chromium.png|5009| Changed|
|Switch.Disabled - Dark Mode.hover.chromium.png|909| Changed|
|Switch.Disabled - RTL.hover.chromium.png|4914| Changed|
|Switch.Disabled - Dark Mode.normal.chromium.png|909| Changed|
|Switch.Disabled - RTL.normal.chromium.png|4914| Changed|
|Switch.Default.hover.chromium.png|5044| Changed|
|Switch.Disabled Checked - Dark Mode.hover.chromium.png|1074| Changed|
|Switch.Default - RTL.normal.chromium.png|5009| Changed|
|Switch.Disabled Checked - Dark Mode.normal.chromium.png|1074| Changed|
|Switch.Disabled Checked.normal.chromium.png|5047| Changed|
|Switch.Default - RTL.hover.chromium.png|5044| Changed|
|Switch.Disabled.hover.chromium.png|4914| Changed|
|Switch.Default - Dark Mode.normal.chromium.png|1019| Changed|
|Switch.Checked - Dark Mode.hover.chromium.png|1169| Changed|
|Switch.Default - Dark Mode.hover.chromium.png|1071| Changed|
|Switch.Disabled.normal.chromium.png|4914| Changed|
|Switch.Disabled Checked.hover.chromium.png|5047| Changed|
|Switch.Label Above - RTL.hover.chromium.png|0| Removed|
|Switch.Label Above - RTL.normal.chromium.png|0| Removed|
|Switch.Label Above.hover.chromium.png|0| Removed|
|Switch.Label Above.normal.chromium.png|0| Removed|
|Switch.Label After - RTL.hover.chromium.png|0| Removed|
|Switch.Label After - RTL.normal.chromium.png|0| Removed|
|Switch.Label After.hover.chromium.png|0| Removed|
|Switch.Label After.normal.chromium.png|0| Removed|
|Switch.Label Before - RTL.hover.chromium.png|0| Removed|
|Switch.Label Before - RTL.normal.chromium.png|0| Removed|
|Switch.Label Before.hover.chromium.png|0| Removed|
|Switch.Label Before.normal.chromium.png|0| Removed|

</details>


  <details><summary><b>Text</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Text.Strikethrough.normal.chromium.png|236| Changed|
|Text.Block.normal.chromium.png|995| Changed|

</details>


  <details><summary><b>TextInput</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|TextInput.Size Large.normal.chromium.png|3| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

File: packages/web-components/src/link/link.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+   * In chromium browsers, if the link is contained within a semantic
+   * text element (`h1`, `h2`, `h3`, `h4`, `h5`, `h6`, `p`) or `fluent-text`,
+   * `:host-context()` ensures inline links are styled appropriately.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `inline`
+   */
+  @attr({ mode: 'boolean' })
+  public inline?: boolean = false;
```
Comment (davatron5000) [inline packages/web-components/src/link/link.ts:39]: fwiw, I assumed `inline` was for controlling display. it took me awhile to figure out it was about inheriting styles when used as phrasing content. can't think of a better word though.

File: packages/web-components/src/anchor-button/anchor-button.ts
Comment (davatron5000) [inline packages/web-components/src/anchor-button/anchor-button.ts:1]: I think for now it's fine, but it seems like `BaseAnchor` and other `Base*` classes may need a home somewhere. As a consumer I wouldn't expect to import `BaseAnchor` from `AnchorButton` (e.g. how it's done in `Link`)

File: packages/web-components/src/link/link.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { attr } from '@microsoft/fast-element';
+import { BaseAnchor } from '../anchor-button/anchor-button.js';
+import { StartEnd } from '../patterns/index.js';
+import { applyMixins } from '../utils/apply-mixins.js';
```
Comment (davatron5000) [inline packages/web-components/src/link/link.ts:4]: I didn't see these used

File: packages/web-components/src/link/link.ts (hunk 2 above)
Comment (davatron5000) [reply packages/web-components/src/link/link.ts:39]: If it's per the design spec, then I think we roll with it until it needs to change. The lack of cross-browser `:host-context()` is frustrating.

Comment (chrisdholt) [reply packages/web-components/src/link/link.ts:39]: Yeah, I'd love to see broader adoption of that, it's a huge benefit for CE's in some scenarios.

File: packages/web-components/src/anchor-button/anchor-button.ts
Comment (chrisdholt) [reply packages/web-components/src/anchor-button/anchor-button.ts:1]: Yeah, I'd agree here - I think once we get some of these abstractions in place we can look for a way to:
1. Better organize them
2. Potentially export them for reuse?

**Review by davatron5000 — APPROVED** [review]

File: packages/web-components/src/link/link.options.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * Link Appearance constants
+ * @public
+ */
+export const LinkAppearance = {
+  subtle: 'subtle',
+} as const;
+
+/**
+ * An Link can be secondary, primary, outline, subtle, transparent
```
Comment (mlijanto) [inline packages/web-components/src/link/link.options.ts:13]: Is this comment accurate? Looks like `LinkAppearance` above only has `subtle` as an option.

File: packages/web-components/src/link/link.template.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { ElementViewTemplate, html, ViewTemplate } from '@microsoft/fast-element';
+import type { Link } from './link.js';
+
+/**
+ * The template for the Button component.
```
Comment (mlijanto) [inline packages/web-components/src/link/link.template.ts:5]: Should this say "Link" instead of "Button?"

Comment (chrisdholt) [reply packages/web-components/src/link/link.template.ts:5]: Yes - I'll push a PR 🗡️

File: packages/web-components/src/link/link.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+   * In chromium browsers, if the link is contained within a semantic
+   * text element (`h1`, `h2`, `h3`, `h4`, `h5`, `h6`, `p`) or `fluent-text`,
+   * `:host-context()` ensures inline links are styled appropriately.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `inline`
+   */
+  @attr({ mode: 'boolean' })
+  public inline?: boolean = false;
```
Comment (mlijanto) [inline packages/web-components/src/link/link.ts:37]: Nit: if this property has a default value then it doesn't need to be marked as optional.

**Pull request merged**